### PR TITLE
Fix chem grenades remaining equipped after in-hand detonation

### PR DESCRIPTION
**Pull request by PrimeNumb**
## About the PR 
Chem grenades were equipped and invisible for a short while after in-hand detonation. Now they're not.

## Why's this needed? 
bugfix

### CODE REVIEW

**Comment by flappybatpal** [comment]
I believe all chem grenades experience this?

https://github.com/goonstation/goonstation/blob/e54949c130fca43afbb55391758eb2ea9ab5293c/code/modules/chemistry/tools/grenades.dm#L204

I experimented with an unequip call but I'm not sure of the implications.

**Comment by PrimeNumb** [comment]
> I believe all chem grenades experience this?
> 
> https://github.com/goonstation/goonstation/blob/e54949c130fca43afbb55391758eb2ea9ab5293c/code/modules/chemistry/tools/grenades.dm#L204
> 
> I experimented with an unequip call but I'm not sure of the implications.

The invisibility is correct behavior (for reasons), but being equipped after detonation is not.

File: code/modules/chemistry/tools/grenades.dm [hunk 1]
```diff
@@ -170,6 +170,8 @@
 			if (src && !src.disposed)
 				a = get_area(src)
 				if(a.sanctuary) return
+				if(user && user.equipped() == src)
+					user.u_equip(src)
```
Comment (ZeWaka) [inline code/modules/chemistry/tools/grenades.dm:174]: Why not just call this irregardless

Comment (PrimeNumb) [reply code/modules/chemistry/tools/grenades.dm:174]: `u_equip()` will call `dropped()` for non-human mobs regardless of if the item was equipped or not, which could lead to undesired behavior. For example: handspiders would be able to bypass their sprint restrictions if I removed this check. It should probably be done inside `u_equip()` instead, but I was reluctant to mess with it at first due to how many things could (potentially) break from it.

**Pull request merged**